### PR TITLE
added I2C driver for TeraRanger One

### PR DESCRIPTION
**Pull request by ndepal**
also see PR PX4/DriverFramework#52

### CODE REVIEW

**Comment by tuofeichen** [comment]
sorry, I'm a little confused, does the trone.cpp in drivers directory not working right now? I thought that's also I2C?

**Comment by mhkabir** [comment]
This is based on the new DriverFramework platform,  so this driver should work on all platforms like RPi, Snapdragon, NuttX, etc.

**Comment by julianoes** [comment]
@mhkabir It won't work on NuttX just yet but sometime in the future we might be able to merge the parallel driver worlds.

File: src/platforms/posix/drivers/df_trone_wrapper/df_trone_wrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	int			_orb_class_instance;
+
+	// perf_counter_t		_range_sample_perf;
+
+};
+
+DfTROneWrapper::DfTROneWrapper(/*enum Rotation rotation*/) :
+	TROne(TRONE_DEVICE_PATH),
+	_range_topic(nullptr),
+	// _param_update_sub(-1),
```
Comment (julianoes) [inline src/platforms/posix/drivers/df_trone_wrapper/df_trone_wrapper.cpp]: I'd say just delete if you don't need it.

File: src/platforms/posix/drivers/df_trone_wrapper/df_trone_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+		PX4_ERR("TROne stop fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfTROneWrapper::_publish(struct range_sensor_data &data)
+{
+	if (!_range_topic)
```
Comment (julianoes) [inline src/platforms/posix/drivers/df_trone_wrapper/df_trone_wrapper.cpp]: Please always put curly brackets for if/for/while.
Also `make check_format` should catch this.

**Comment by julianoes** [comment]
Reviewed this and https://github.com/PX4/DriverFramework/pull/52. Have a look at the comments.

**Comment by LorenzMeier** [comment]
@julianoes Can you manage the integration with Nicolas?

**Comment by julianoes** [comment]
All looked, good, thanks! Rebased to master.

**Pull request closed without merging**